marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/data-services/mapping/getNewDocument.sjs

```diff
@@ -0,0 +1,42 @@
+/**
+ Copyright (c) 2020 MarkLogic Corporation
+
+ Licensed under the Apache License, Version 2.0 (the 'License');
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an 'AS IS' BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+'use strict';
+
+xdmp.securityAssert('http://marklogic.com/data-hub/privileges/read-mapping', 'execute');
+
+var stepName, uri;
+
+const rtn = {
+  data: null,
+  namespaces: {},
+  format: null
+}
+
+const doc = cts.doc(uri);
```

Comment:
If the doc isn't found, we should immediately throw an error. Use ds-util.sjs, and add a new function to it that we haven't needed yet, but we need now - ds.throwNotFound("Could not find a document with URI: " + uri);

Reply:
Addressed.  Will be present in updated/next PR.